src/download/chunk_statistics.cc

```diff
@@ -145,9 +145,9 @@ ChunkStatistics::received_have_chunk(PeerChunks* pc, uint32_t index, uint32_t le
       
       m_complete++;
       m_accounted--;
-      
-      for (iterator itr = base_type::begin(), last = base_type::end(); itr != last; ++itr)
-	*itr -= 1;
+
+      for (auto& c : *this)
+        const_cast<uint8_t&>(c) -= 1;
```

Comment:
Using const_cast looks wrong, should access the base_type instead.

Reply:
I had trouble getting this to compile without const_cast.

Reply:
Yeah this compiles:

```
--- a/src/download/chunk_statistics.cc
+++ b/src/download/chunk_statistics.cc
@@ -146,8 +146,8 @@ ChunkStatistics::received_have_chunk(PeerChunks* pc, uint32_t index, uint32_t le
       m_complete++;
       m_accounted--;
 
-      for (auto& c : *this)
-        const_cast<uint8_t&>(c) -= 1;
+      for (auto c : *this)
+        c -= 1;
     }
 
   } else {

```

but this looks wrong.

Reply:
You need to cast 'this' to base_type.

Reply:
Gave up and converted to std::transform.